### PR TITLE
fix(components): [switch] fix type error

### PR DESCRIPTION
**Pull request by Chuck-Lau**
#6114 

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/switch/__tests__/switch.test.tsx [hunk 1]
```diff
@@ -185,7 +184,7 @@ describe('Switch.vue', () => {
     const coreWrapper = wrapper.find('.el-switch__core')
     const switchWrapper: VueWrapper<SwitchInstance> =
       wrapper.findComponent(Switch)
-    const switchVm = switchWrapper.vm
+    const switchVm: any = switchWrapper.vm
```
Comment (tolking) [inline packages/components/switch/__tests__/switch.test.tsx:187]: Is it possible to avoid use the any?

Comment (Tsong-LC) [reply packages/components/switch/__tests__/switch.test.tsx:187]: I've tried, but no way not to use any.

File: packages/components/switch/__tests__/switch.test.tsx [hunk 2]
```diff
@@ -111,8 +110,8 @@ describe('Switch.vue', () => {
   test('change event', async () => {
     const target = ref(1)
     const value = ref(true)
-    const handleChange = (val: boolean) => {
-      target.value = val
+    const handleChange = (val: string | number | boolean) => {
+      target.value = val as number
```
Comment (holazz) [inline packages/components/switch/__tests__/switch.test.tsx:114]: I think passing a generic parameter to target ref is better than using type assertion.

```ts
const target = ref<string | number | boolean>(1)
```

File: packages/components/switch/__tests__/switch.test.tsx [hunk 3]
```diff
@@ -238,7 +237,7 @@ describe('Switch.vue', () => {
     const value = ref(true)
     const loading = ref(false)
     const asyncResult = ref('error')
-    const beforeChange = () => {
+    const beforeChange: () => Promise<boolean> | boolean = () => {
```
Comment (holazz) [inline packages/components/switch/__tests__/switch.test.tsx:240]: Here we can use a more specific type.

```ts
const beforeChange = () => {
  return new Promise<boolean>((resolve, reject) => {
   	...
  })
}
```

File: packages/components/switch/__tests__/switch.test.tsx (hunk 1 above)
Comment (holazz) [reply packages/components/switch/__tests__/switch.test.tsx:187]: Maybe expose `checked` from the `el-switch`?

Comment (tolking) [reply packages/components/switch/__tests__/switch.test.tsx]: It seems that removing `:any` does not cause a type error.

**Review by xiaoxian521 — APPROVED** [review]

**Pull request merged**